From rajesh.borundia@qlogic.com  Thu Jun 23 22:49:06 2011
From: Rajesh Borundia <rajesh.borundia@qlogic.com>
Date: Thu, 16 Dec 2010 22:59:02 +0000
Subject: [PATCH 168/187] qlcnic: reset pci function unconditionally during
 probe
Git-commit: 1dc0f3c54ce1df957f99c17b145488fd03eb1a59
Patch-mainline: v2.6.38-rc1
References: bnc#698272, FATE#311468

Some boot code drivers dont have cleanup routine, so pci function
remains in unknown state prior to driver load. So during driver load
issue FLR unconditionally.

Update driver version to 5.0.14.

Signed-off-by: Rajesh Borundia <rajesh.borundia@qlogic.com>
Signed-off-by: Amit Kumar Salecha <amit.salecha@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/qlcnic/qlcnic.h      |    4 ++--
 drivers/net/qlcnic/qlcnic_main.c |    5 +----
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/drivers/net/qlcnic/qlcnic.h b/drivers/net/qlcnic/qlcnic.h
index 527f6cd..96de41e 100644
--- a/drivers/net/qlcnic/qlcnic.h
+++ b/drivers/net/qlcnic/qlcnic.h
@@ -35,8 +35,8 @@
 
 #define _QLCNIC_LINUX_MAJOR 5
 #define _QLCNIC_LINUX_MINOR 0
-#define _QLCNIC_LINUX_SUBVERSION 13
-#define QLCNIC_LINUX_VERSIONID  "5.0.13"
+#define _QLCNIC_LINUX_SUBVERSION 14
+#define QLCNIC_LINUX_VERSIONID  "5.0.14"
 #define QLCNIC_DRV_IDC_VER  0x01
 #define QLCNIC_DRIVER_VERSION  ((_QLCNIC_LINUX_MAJOR << 16) |\
 		 (_QLCNIC_LINUX_MINOR << 8) | (_QLCNIC_LINUX_SUBVERSION))
diff --git a/drivers/net/qlcnic/qlcnic_main.c b/drivers/net/qlcnic/qlcnic_main.c
index ad4581b..03b2095 100644
--- a/drivers/net/qlcnic/qlcnic_main.c
+++ b/drivers/net/qlcnic/qlcnic_main.c
@@ -1472,7 +1472,6 @@ qlcnic_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 	uint8_t revision_id;
 	uint8_t pci_using_dac;
 	char brd_name[QLCNIC_MAX_BOARD_NAME_LEN];
-	u32 val;
 
 	err = pci_enable_device(pdev);
 	if (err)
@@ -1534,9 +1533,7 @@ qlcnic_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 	if (err)
 		goto err_out_iounmap;
 
-	val = QLCRD32(adapter, QLCNIC_CRB_DRV_ACTIVE);
-	if (QLC_DEV_CHECK_ACTIVE(val, adapter->portnum))
-		adapter->flags |= QLCNIC_NEED_FLR;
+	adapter->flags |= QLCNIC_NEED_FLR;
 
 	err = adapter->nic_ops->start_firmware(adapter);
 	if (err) {
-- 
1.6.3.3

